packages/app-mobile/components/NoteEditor/SearchPanel.tsx

```diff
@@ -0,0 +1,335 @@
+
+	searchText: '',
+	replaceText: '',
+	dialogVisible: false,
+};
+
+export interface SearchPanelProps {
+    searchControl: SearchControl;
+    searchState: SearchState;
+    editorSettings: EditorSettings;
+}
+
+const ActionButton = (
+	props: { styles: any; iconName: string; title: string; onPress: (()=> void) }
+) => {
+	return (
+		<TouchableOpacity
+			style={props.styles.button}
+			onPress={props.onPress}
+
+			accessibilityLabel={props.title}
+			accessibilityRole='button'
+		>
+			<MaterialCommunityIcon name={props.iconName} style={props.styles.buttonText}/>
+		</TouchableOpacity>
+	);
+};
+
+const ToggleButton = (
+	props: { styles: any; iconName: string; title: string; active: boolean; onToggle: (()=> void) }
+) => {
+	const active = props.active;
+
+	return (
+		<TouchableOpacity
+			style={{
+				...props.styles.toggleButton,
+				...(active ? props.styles.toggleButtonActive : {}),
+			}}
+			onPress={props.onToggle}
+
+			accessibilityState={{
+				checked: props.active,
+			}}
+			accessibilityLabel={props.title}
+			accessibilityRole='switch'
+		>
+			<MaterialCommunityIcon name={props.iconName} style={
+				active ? props.styles.activeButtonText : props.styles.buttonText
+			}/>
+		</TouchableOpacity>
+	);
+};
+
+
+const useStyles = (theme: any) => {
+	return useMemo(() => {
+		const buttonStyle = {
+			width: buttonSize,
+			height: buttonSize,
+			backgroundColor: theme.backgroundColor4,
+			alignItems: 'center',
+			justifyContent: 'center',
+			flexShrink: 1,
+		};
+		const buttonTextStyle = {
+			color: theme.color4,
+			fontSize: 30,
+		};
+
+		return StyleSheet.create({
+			button: buttonStyle,
+			toggleButton: {
+				...buttonStyle,
+			},
+			toggleButtonActive: {
+				...buttonStyle,
+				backgroundColor: theme.backgroundColor3,
+			},
+			input: {
+				flexGrow: 1,
+				height: buttonSize,
+				backgroundColor: theme.backgroundColor4,
+				color: theme.color4,
+			},
+			buttonText: buttonTextStyle,
+			activeButtonText: {
+				...buttonTextStyle,
+				color: theme.color4,
+			},
+			text: {
+				color: theme.color,
+			},
+			labeledInput: {
+				flexGrow: 1,
+				flexDirection: 'row',
+				alignItems: 'center',
+				justifyContent: 'center',
+				marginLeft: 10,
+			},
+		});
+	}, [theme]);
+};
+
+export const SearchPanel = (props: SearchPanelProps) => {
+	const placeholderColor = props.editorSettings.themeData.color3;
+	const styles = useStyles(props.editorSettings.themeData);
+
+	const [showingAdvanced, setShowAdvanced] = useState(false);
+
+	const state = props.searchState;
+	const control = props.searchControl;
+
+	const updateSearchState = (changedData: any) => {
+		const newState = Object.assign({}, state, changedData);
+		control.setSearchState(newState);
+	};
+
+	// Creates a TextInut with the given parameters
+	const createInput = (
+		{ placeholder, value, onChange, autoFocus = false }:
+			{ placeholder: string; value: string; onChange: (text: string)=> void; autoFocus?: boolean }
```

Comment:
Again please don't use this pattern.

- If an object interface is used more than once, define a properly named interface

- If it's used only once, there's a chance you don't need a interface. In this particular case, just used regular function arguments. That way you don't need to repeat the names twice over two lines.

Reply:
Good catch! (I wrote this code a while ago and forgot to update it.)